### PR TITLE
Make logging and behavior stack creation lazy.

### PR DESCRIPTION
**Pull request by adamhathcock**
Covers https://github.com/AsynkronIT/protoactor-dotnet/issues/108

### CODE REVIEW

File: src/Proto.Actor/LocalContext.cs [hunk 1]
```diff
@@ -124,18 +134,26 @@ public PID SpawnNamed(Props props, string name)
 
         public void SetBehavior(Receive receive)
         {
-            _behavior.Clear();            
+            _behavior = null;
```
Comment (adamhathcock) [inline src/Proto.Actor/LocalContext.cs]: Not sure if clear is better than setting it to null.  I guess I should just clear it if it's already made.

Comment (rogeralsing) [inline src/Proto.Actor/LocalContext.cs]: I think this is good, if we can clear out objects that are no longer used, we should do so 👍

**Comment by rogeralsing** [comment]
Looks good!

File: src/Proto.Actor/LocalContext.cs [hunk 2]
```diff
@@ -124,18 +134,26 @@ public PID SpawnNamed(Props props, string name)
...
+            }
             _behavior.Push(_receive);
             _receive = receive;
         }
 
         public void PopBehavior()
         {
+            if(_behavior == null)
+            {
+                return;
```
Comment (rogeralsing) [inline src/Proto.Actor/LocalContext.cs]: Thinking about it, we should probably throw this ´throw new Exception("Can not unbecome actor base behavior");` when _behavior is null, as the meaning is the same as the later statements.

**Pull request merged**